### ISSUE TITLE
Support for ignoring specific doc sections

### ISSUE THREAD

**Issue by tpansino** [darkriszty/MarkdownTablePrettify-VSCodeExt#56]
I use [terraform-docs](https://github.com/terraform-docs/terraform-docs) to generate and maintain Markdown documentation for some Terraform modules, but the format of the generated tables doesn't match what this extension expects. This means that, when I run these two tools on my docs via automation, they fight each other.

Would it be possible to add support to this tool to ignore certain blocks of input? I'm thinking something similar to [what prettier does](https://prettier.io/docs/en/ignore.html#range-ignore):

```markdown
# This should be formatted

| Column A | Column B |
|----------|----------|
| Foo      | Bar      |

# This should not be formatted

<!-- markdown-table-prettify-ignore START -->
| Column A | Column B |
|---|---|
| Foo | Bar |
<!-- markdown-table-prettify-ignore END -->

# This should be formatted again

| Column A | Column B |
|----------|----------|
| Foo      | Bar      |
```

FWIW I'm not a skilled JS/TS developer, but I can probably contribute something if a maintainer can give me an explanation of which components would need to be touched.

**Comment by darkriszty** [comment]
Thanks for opening the issue, I did not know about this use case 🙂 I will take a look to see how this can be added.

**Comment by darkriszty** [comment]
I've added support to mark ignored sections of documents:
 * The markers are `<!-- markdown-table-prettify-ignore-start -->` and `<!-- markdown-table-prettify-ignore-end -->`.
 * Since there is no lexer or parser involved, the markers have to be precise and must be on dedicated lines, without any additional characters or text before or after.
 * If an ignore-start is in a middle of a valid table, then it will be prettyfied only up to that ignore-start. A later ignore-end in the same table will not "resume" the formatting, as the table is considered to have ended with the ignore-start.
 * The ignore-end is optional.
 * The `Format Selection` only considers the markers if they are inside the selection.
 
These changes are available in version 3.4.0 of the extension.
